python/tskit/util.py

```diff
@@ -592,3 +592,33 @@ def random_nucleotides(length: numbers.Number, *, seed: Union[int, None] = None)
     encoded_nucleotides = np.array(list(map(ord, "ACTG")), dtype=np.int8)
     a = rng.choice(encoded_nucleotides, size=int(length))
     return a.tobytes().decode("ascii")
+
+
+#
+# Miscellaneous auxiliary methods.
+#
+
+
+def allele_remap(alleles_from, alleles_to):
+    """
+    Returns an index map from the elements in one list (alleles_from)
+    to the elements of another list (alleles_to).
+
+    If some elements in alleles_from are not in alleles_to,
+    then indices outside of alleles_to are used.
+
+    :param list alleles_from: A list of alleles to map from.
```

Comment:
The current type requirement for this function isn't `list` but `Sequence` as things like strings, tuples and bytes support the `.index` method.
I wonder if we should be supporting numpy arrays as a possible input to this function? I know `Variant.alleles` returns a `tuple`, so maybe we don't need to.